src/Hooks/HookManager.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Carnival\Hooks;
+
+use Carnival\Hooks\Topic;
+use Illuminate\Support\Collection;
+
+class HookManager {
+
+    protected Collection $hooks;
+
+    public function __construct() {
+        $this->hooks = collect();
+    }
+
+    function add(string $eventClass, callable $callback) : void {
+        data_set($this->hooks, $eventClass, data_get($this->hooks, $eventClass, collect())->push($callback));
+    }
+    
+    function remove(string $eventClass) : void {    
```

Comment:
Mentioned this on Wednesday, I think. Do we really have a use case for a `remove` method here? I can't imagine it being useful. Can you think of anything?

Reply:
Not really, I have no issues removing it. And if we find a reason to re-implement it it is relatively simple.